manifests/job.pp

```diff
@@ -34,11 +37,16 @@
   $jobname  = $title,
   $enabled  = 1,
   $ensure   = 'present',
+  $folder   = undef,
   $difftool = '/usr/bin/diff -b -q',
 ){
   include ::jenkins::cli
 
-  validate_string($difftool)
+  validate_string($difftool, $folder)
+
+  if $folder and (!defined(Jenkins::Plugin['cloudbees-folder'])) {
+    fail('cloudbees-folder plugin is required for the \'folder\' parameter.')
+  }
```

Comment:
@jhoblitt  having the plugin install automatically when someone uses the folder parameter without declaring it elsewhere causes a dependency loop which I didn't manage to solve.

So I think it's easier to just fail and inform the user to specifically add it themselves if they want to use the folder feature.

If they don't use it to begin with, everything works like before.